jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/java/org/jbpm/console/ng/cm/client/comments/CaseCommentsViewImpl.java

```diff
@@ -0,0 +1,151 @@
+    public Span newCommentLabel;
+
+    @Inject
+    @DataField
+    public Button addCommentButton;
+
+
+    @Inject
+    private javax.enterprise.event.Event<NotificationEvent> notification;
+
+    @Inject
+    private ManagedInstance<CaseCommentItemView> provider;
+
+    private CaseCommentsPresenter presenter;
+
+
+    @Override
+    public void init(final CaseCommentsPresenter presenter) {
+        this.presenter = presenter;
+    }
+
+    @Override
+    public void removeAllComments() {
+        clearCommentInput();
+        removeAllChildren(comments);
+    }
+
+    @Override
+    public void clearCommentInput() {
+        newCommentTextArea.setValue("");
+    }
+
+    @Override
+    public void addComment(boolean editing, String editActionLabel, final String commentId, final String author,
+                           final String commentText, final Date commentAddedAt, final CaseCommentsPresenter.CaseCommentAction... actions) {
+        addCommentView( editing, editActionLabel, commentId, author, commentText, commentAddedAt, "fa-comment-o", actions);
+    }
+
+    private void addCommentView(boolean editing, String editActionLabel, final String commentId, final String author,
+                                final String commentText, final Date commentAddedAt, final String iconType,
+                                final CaseCommentsPresenter.CaseCommentAction... actions) {
+        final CaseCommentItemView commentItemView = provider.get();
+        commentItemView.setCommentAuthor(author);
+        commentItemView.setCommentText(commentText);
+        commentItemView.setCommentAddedAt(DateUtils.getDateStr(commentAddedAt));
+
+        commentItemView.setIconType(iconType);
+        commentItemView.setEditMode(editing);
+
+        if (editActionLabel != null) {
+            for (CaseCommentsPresenter.CaseCommentAction action : actions) {
+                commentItemView.addAction(action);
+            }
+
+            commentItemView.addAction(new CaseCommentsPresenter.CaseCommentAction() {
+
+                @Override
+                public String label() {
+                    return editActionLabel;
+                }
+
+                @Override
+                public void execute() {
+                    if (!editing) {
+                        presenter.setCurrentUpdatedCommentId(commentId);
+                        presenter.refreshComments();
+                        commentItemView.setEditMode(true);
+                    } else {
+                        presenter.setCurrentUpdatedCommentId("");
+                        presenter.updateCaseComment(commentItemView.getUpdatedComment(), commentId);
+                        commentItemView.setEditMode(false);
+                    }
+                }
+            });
+        }
+        comments.appendChild(commentItemView.getElement());
+    }
+
+
+    @Override
+    public void displayNotification( String text ) {
+        notification.fire( new NotificationEvent( text ) );
```

Comment:
Instead of using event notifications for validation error, I would suggest using the standard form state error for the comment input.